docs/contributing.md

```diff
@@ -2,37 +2,92 @@
 
 ## Mailing list
 
-The main communication tool we use is a mailing list:
+The main communication tool in use is the Yocto Project mailing list:
+
 * <yocto@yoctoproject.org>
 * <https://lists.yoctoproject.org/listinfo/yocto>
 
-Feel free to ask any kind of questions but always prepend your email subject
-with "[meta-raspberrypi]". This is because we use the 'yocto' mailing list and
-not a perticular 'meta-raspberrypi' mailing list.
+Feel free to ask any kind of questions but please always prepend your email
+subject with `[meta-raspberrypi]` as this is the global *Yocto* mailing
+list and not a dedicated *meta-raspberrypi* mailing list.
+
+## Formatting patches
+
+First and foremost, all of the contributions to the layer must be compliant
+with the standard openembedded patch guidelines:
+
+* <http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines>
+
+In summary, your commit log messages should be formatted as follows:
+
+    <layer-component>: <short summary of changes>
+
+    <more detailed description of changes and considerations>
+
+    Signed-off-by: Your Name <yourname@youremail.com>
+
+The `<layer-component>` is the layer component name that your changes affect.
+It is important that you choose it correctly. A simple guide for selecting a
+a good component name is the following:
+
+* For changes that affect the *layer itself* such as adding new recipes or
+global functionality, please use `meta-raspberrypi` as the component name.
+* For changes that affect *layer recipes*, please just use the **base name**
+of the recipe as the component name. For example, use `omxplayer` instead of
+`omxplayer_git.bb`. This also applies when adding extra files to recipes.
+* For changes that affect the *layer documentation*, please just use `docs`
+as the component name.
+* For changes that affect *other files*, i.e. under the `conf` directory,
+please use the full path as the component name, e.g. `conf/layer.conf`.
```

Comment:
I think this should actually be documented in general openembedded document and rpi layer should point to it. or atleast ensure they are in agreement

Reply:
I agree, however the main documentation guidelines are a bit broad. Their own example is:

    Short log / Statement of what needed to be changed.
    
    (Optional pointers to external resources, such as defect tracking)
    
    The intent of your change.
    
    (Optional, if it's not clear from above) how your change resolves the
    issues in the first part.
    
    Tag line(s) at the end.

So I think in the end is up to the individual maintainers to fine-tune how they want their log formatting in their layers. In this case, after the chat with @agherzan we agreed the proposed formatting was handy and made sense to keep things tidy.

We can always try to submit these guidelines upstream, but I think for the moment is good to have it here as the low-hanging fruit for our users. Also, I don't want to turn this into an opinion-wars :)

Reply:
@kraj I'd love to have a unified guideline but I'm as well worried about the time we need to spend bringing this to oe and discussing a solution at that level. Let's for now fix for us. And if anyone has the time to pursue this to OE, I'll support it as much as I can.

Reply:
thats fine, I think, I am interested in keeping it in sync so folks are not confused.

https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines

and

http://www.openembedded.org/wiki/How_to_submit_a_patch_to_OpenEmbedded

do have detailed prescriptions, now question is should they be improved and simply referred to here with additional github related stuff we talked about or should it rework whats said
in those docs.

Reply:
Yes, the majority of the new content I added comes essentially from these two links as they were in the documentation before I edited it, so I guess is fine :)
The only real addition I would say is the explicit statement of how the commit log messages are expected to be received for this layer (the new guidelines). This essentially is the second option you state (*"should it rework whats said in those docs"*).
I agree that referencing upstream documentation is desirable, but experience also says that users often need to be explicitly and directly told precisely how to do stuff, and in this case is not a big deal to rephrase upstream documentation down here, I think :)

Reply:
there is a small isssue with duplicating information. They need to be kept in sync, the OE documents are living documents and they get updated. So we might be signing up for this sync operation instead if we link to those docs then we dont have to worry about it.

Reply:
I agree with that (the commitment of syncing), yes. However I still believe that in this case we are not really changing the upstream docs, we are just specifying more clearly and with examples how things are expected, e.g. the component namings, github keywords, the signed-off tag.

The upstream documentation is broad about that and does not enforce these things so I honestly don't think the improvements in the documentation here are going to get obsolete any time soon.

What you suggest me to do to get this PR going?

Reply:
Hi @kraj, a gently reminder to let me know how you would like me to change the PR to get it done :) Thanks!

Reply:
Merry Christmas, @kraj !